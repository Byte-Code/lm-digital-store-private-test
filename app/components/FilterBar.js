import React, { Component, PropTypes } from 'react';
import ImmutablePropTypes from 'react-immutable-proptypes';
import styled from 'styled-components';
import Dialog from 'material-ui/Dialog';
import AddIcon from 'material-ui/svg-icons/content/add-circle-outline';

import FilterDialog from './FilterDialog';
import ActiveFilters from './ActiveFilters';

const Wrapper = styled.div`
  display: flex;
  height: 71px;
`;

export const Button = styled.div`
  width: 190px;
  display: flex;
  align-items: center;
  justify-content: center;
  background: rgba(51, 51, 51, 0.8);
  cursor: pointer;
  &>p {
    font-size: 20px;
    line-height: 32px;
    font-family: LeroyMerlinSans Light;
    margin-left: 12px;
    color: #fff;
  }
`;

export default class FilterBar extends Component {
  static propTypes = {
    filterGroups: ImmutablePropTypes.list.isRequired,
    resetFilters: PropTypes.func.isRequired,
    applyFilters: PropTypes.func.isRequired,
    toggleFilter: PropTypes.func.isRequired,
    filterMap: ImmutablePropTypes.map.isRequired,
    toggleAvailability: PropTypes.func.isRequired
  }

  constructor(props) {
    super(props);
    this.state = {
      open: false
    };
  }

  handleOpen = () => {
    this.setState({ open: true });
  }

  handleClose = () => {
    this.setState({ open: false });
  }

  // TODO separate dialog logic into dialog component?
  render() {
    const {
      applyFilters,
      filterGroups,
      filterMap,
      resetFilters,
      toggleFilter,
      toggleAvailability
    } = this.props;

    if (filterGroups.isEmpty()) {
      return null;
    }

    return (
      <Wrapper>
        <Button onClick={this.handleOpen}>
          <AddIcon color="#fff" style={{ height: 30, width: 30 }} />
          <p>Pi√∫ filtri</p>
        </Button>
        <ActiveFilters
          filterMap={filterMap}
          filterGroups={filterGroups}
          resetFilters={resetFilters}
          toggleFilter={toggleFilter}
          toggleAvailability={toggleAvailability}
          handleOpen={this.handleOpen}
        />
        <Dialog
          modal={false}
          onRequestClose={this.handleClose}
          open={this.state.open}
          contentStyle={{ width: '100%', maxWidth: 'none' }}
          bodyStyle={{ padding: 0, background: 'rgba(51, 51, 51, 0.8)' }}
        >
          <FilterDialog
            filterGroups={this.props.filterGroups}
            handleClose={this.handleClose}
            applyFilters={applyFilters}
            filterMap={filterMap}
          />
        </Dialog>
      </Wrapper>
    );
  }
}
