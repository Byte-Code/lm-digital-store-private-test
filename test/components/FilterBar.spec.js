import React from 'react';
import { fromJS } from 'immutable';
import { shallow } from 'enzyme';

import FilterBar, { Button } from '../../app/components/FilterBar';

describe('filterBar', () => {
  const filterMap = fromJS({});
  const resetFilters = () => {};
  const applyFilters = () => {};
  const toggleFilter = () => {};
  const toggleAvailability = () => {};

  it('should return null when filterGroups are empty', () => {
    const filterGroups = fromJS([]);
    const result = shallow(
      <FilterBar
        resetFilters={resetFilters}
        applyFilters={applyFilters}
        toggleFilter={toggleFilter}
        toggleAvailability={toggleAvailability}
        filterMap={filterMap}
        filterGroups={filterGroups}
      />
    );
    expect(result).toMatchSnapshot();
  });

  it('should initially render with a closed dialog', () => {
    const filterGroups = fromJS([1, 2, 3]);
    const result = shallow(
      <FilterBar
        resetFilters={resetFilters}
        applyFilters={applyFilters}
        toggleFilter={toggleFilter}
        toggleAvailability={toggleAvailability}
        filterMap={filterMap}
        filterGroups={filterGroups}
      />
    );
    expect(result.state('open')).toBe(false);
    expect(result).toMatchSnapshot();
  });

  it('should open the dialog when the button is clicked', () => {
    const filterGroups = fromJS([1, 2, 3]);
    const result = shallow(
      <FilterBar
        resetFilters={resetFilters}
        applyFilters={applyFilters}
        toggleFilter={toggleFilter}
        toggleAvailability={toggleAvailability}
        filterMap={filterMap}
        filterGroups={filterGroups}
      />
    );
    result.find(Button).simulate('click');
    expect(result.state('open')).toBe(true);
    expect(result).toMatchSnapshot();
  });
});
