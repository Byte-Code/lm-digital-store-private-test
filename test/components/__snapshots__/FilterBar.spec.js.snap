// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filterBar should initially render with a closed dialog 1`] = `
<styled.div>
  <styled.div
    onClick={[Function]}
  >
    <ContentAddCircleOutline
      color="#fff"
      style={
        Object {
          "height": 30,
          "width": 30,
        }
      }
    />
    <p>
      Piú filtri
    </p>
  </styled.div>
  <ActiveFilters
    filterGroups={
      Array [
        1,
        2,
        3,
      ]
    }
    filterMap={Object {}}
    handleOpen={[Function]}
    resetFilters={[Function]}
    toggleAvailability={[Function]}
    toggleFilter={[Function]}
  />
  <Dialog
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    bodyStyle={
      Object {
        "background": "rgba(51, 51, 51, 0.8)",
        "padding": 0,
      }
    }
    contentStyle={
      Object {
        "maxWidth": "none",
        "width": "100%",
      }
    }
    modal={false}
    onRequestClose={[Function]}
    open={false}
    repositionOnUpdate={true}
  >
    <FilterDialog
      applyFilters={[Function]}
      filterGroups={
        Array [
          1,
          2,
          3,
        ]
      }
      filterMap={Object {}}
      handleClose={[Function]}
    />
  </Dialog>
</styled.div>
`;

exports[`filterBar should open the dialog when the button is clicked 1`] = `
<styled.div>
  <styled.div
    onClick={[Function]}
  >
    <ContentAddCircleOutline
      color="#fff"
      style={
        Object {
          "height": 30,
          "width": 30,
        }
      }
    />
    <p>
      Piú filtri
    </p>
  </styled.div>
  <ActiveFilters
    filterGroups={
      Array [
        1,
        2,
        3,
      ]
    }
    filterMap={Object {}}
    handleOpen={[Function]}
    resetFilters={[Function]}
    toggleAvailability={[Function]}
    toggleFilter={[Function]}
  />
  <Dialog
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    bodyStyle={
      Object {
        "background": "rgba(51, 51, 51, 0.8)",
        "padding": 0,
      }
    }
    contentStyle={
      Object {
        "maxWidth": "none",
        "width": "100%",
      }
    }
    modal={false}
    onRequestClose={[Function]}
    open={true}
    repositionOnUpdate={true}
  >
    <FilterDialog
      applyFilters={[Function]}
      filterGroups={
        Array [
          1,
          2,
          3,
        ]
      }
      filterMap={Object {}}
      handleClose={[Function]}
    />
  </Dialog>
</styled.div>
`;

exports[`filterBar should return null when filterGroups are empty 1`] = `null`;
